iOS Client/TabController/AppCoordinator.swift

```diff
@@ -43,6 +44,51 @@ class AppCoordinator: NSObject, Coordinator, UITabBarControllerDelegate, StoreSu
             self.addChildCoordinator(childCoordinator: coord)
             return coord.rootViewController
         })
+        promptForNotifications()
+    }
+    
+    func promptForNotifications() {
+        let center = UNUserNotificationCenter.current()
+        
+        center.requestAuthorization(options: [.alert, .badge, .sound]) { granted, error in
+            if granted {
+                MBStore().syncDevotions { devotions, error in
+                    if error != nil {
+                        print("Could not sync devotions and schedule devotions")
+                        print(error)
+                    } else if devotions != nil {
+                        center.delegate = self
+                        self.scheduleNotifications(devotions: devotions!)
+                    }
+                }
+            } else {
+                print(error)
+            }
+        }
+    }
+    
+    func scheduleNotifications(devotions: [MBDevotion]) {
+        print("scheduling notifications")
+        let center = UNUserNotificationCenter.current()
+        center.removeAllPendingNotificationRequests()
+        devotions.forEach { devotion in
+            guard let devotionDay = Formatters.devotionDateFormatter.date(from: devotion.date), let calendar = Formatters.calendar else {
+                return
+            }
+            var dateComponents = DateComponents()
+            dateComponents.hour = 10
+            dateComponents.minute = 30
+            dateComponents.year = calendar.component(NSCalendar.Unit.year, from: devotionDay)
+            dateComponents.month = calendar.component(NSCalendar.Unit.month, from: devotionDay)
+            dateComponents.day = calendar.component(NSCalendar.Unit.day, from: devotionDay)
```

Comment:
let's move all this logic to a derived property on MBDevotion class. It could be an optional property of type DateComponents called dateComponents or something, and then here we could just access that lazy property and continue if it's nil or keep going if it's not.

Reply:
we need to move a lot of this boiler plate out of this function